posts/kotlin-like-java-but-better-2.md

```diff
@@ -0,0 +1,255 @@
+@RequestMapping("/api")
+public class PersonController {
+
+    PersonService personService = new PersonService();
+
+    @GetMapping("/ping")
+    public ResponseEntity<String> ping() {
+        return new ResponseEntity<>("pong", HttpStatus.OK);
+    }
+
+    @GetMapping("/people")
+    @ResponseBody
+    public  ResponseEntity<ArrayList<Person>> getPeople() {
+        return new ResponseEntity<>(personService.getPeople(), HttpStatus.OK);
+    }
+
+    @PostMapping("/people/add")
+    public ResponseEntity<String> addPerson(
+            @RequestBody Person person
+    ) {
+        try {
+            return new ResponseEntity<>(personService.addPerson(person).toString(), HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Not a valid person object", HttpStatus.BAD_REQUEST);
+        }
+    }
+}
+```
+Our controller class is fairly basic, without very much we can change or improve as we convert to Kotlin. Notice that in our Kotlin Controller the `person` argument in the `addPerson()` function is not nullable. This is intentional, as Spring will automatically respond with a `400 Bad Request` if a Person object is not passed in the body of the request, so we can always assume that the body request exists. We have also made no changes to any of the Spring annotations, further showcasing the interoperability support that is offered by Spring.
+
+### PersonController.kt
+```kotlin
+@Controller
+@RequestMapping("/api")
+class PersonController {
+    val personService = PersonService()
+
+    @GetMapping("/ping")
+    fun ping(): ResponseEntity<String> { return ResponseEntity("pong", HttpStatus.OK) }
+
+    @GetMapping("/people")
+    @ResponseBody
+    fun getPeople(): ResponseEntity<ArrayList<Person>> {
+        return ResponseEntity(personService.people, HttpStatus.OK)
+    }
+
+    @PostMapping("/people/add")
+    fun addPerson(
+        @RequestBody person: Person
+    ): ResponseEntity<String> {
+        return ResponseEntity(personService.addPerson(person).toString(), HttpStatus.OK)
+    }
+}
+```
+We can run our tests one last time to ensure that our app still works as expected, and then we can sit back and admire our work!
+
+# Wrap up
+By converting our app from Java to Kotlin, we have been able to see some of the features Kotlin that kotlin offers, such as: Data Classes, Java interoperability, and Spring compatibility. We were also able to make the code in our app easier to read by reducing boilerplate code while maintaining a very similar structure. 
+
+Want to learn more about Kotlin? Check out these resources:
+- [Getting Started With Kotlin](https://kotlinlang.org/docs/getting-started.html)
+- [Kotlin Koans](https://play.kotlinlang.org/koans/overview) - A set of exercises to introduce someone familiar with Java to Kotlin syntax and patterns  
+- [Spring Boot Kotlin Tutorial](https://spring.io/guides/tutorials/spring-boot-kotlin/) - Get started creating a Spring Boot project completely in Kotlin.
```

Comment:
Good stuff!  Only thing I would go back and do is delete some of the exclamation points because Kotlin ain't that exciting ( ͡~ ͜ʖ ͡°)

Reply:
I'll tone it down :(